UnitsNet/QuantityTypeConverter.cs

```diff
@@ -0,0 +1,205 @@
+        /// <param name="context">An System.ComponentModel.ITypeDescriptorContext that provides a format context.</param>
+        /// <param name="sourceType">A System.Type that represents the type you want to convert from.</param>
+        /// <returns>true if this converter can perform the conversion; otherwise, false.</returns>
+        public override bool CanConvertFrom(ITypeDescriptorContext context, Type sourceType)
+        {
+            return (sourceType == typeof(string)) || base.CanConvertFrom(context, sourceType);
+        }
+
+        private static TAttribute GetAttribute<TAttribute>(ITypeDescriptorContext context) where TAttribute : UnitAttributeBase
+        {
+            TAttribute attribute = null;
+            AttributeCollection ua = context?.PropertyDescriptor.Attributes;
+
+            attribute = (TAttribute)ua?[typeof(TAttribute)];
+
+            if(attribute != null)
+            {
+                QuantityType expected = default(TQuantity).Type;
+                QuantityType actual = Quantity.From(1, attribute.UnitType).Type;
+                if(expected != actual)
+                {
+                    throw new InvalidOperationException($"The specified UnitType:'{attribute.UnitType}' dose not match QuantityType:'{expected}'");
+                }
+            }
+
+            return attribute;
+        }
+
+        /// <summary>
+        ///     Converts the given object to the type of this converter, using the specified context and culture information.
+        /// </summary>
+        /// <param name="context">An System.ComponentModel.ITypeDescriptorContext that provides a format context.</param>
+        /// <param name="culture">The System.Globalization.CultureInfo to use as the current culture.</param>
+        /// <param name="value">The System.Object to convert.</param>
+        /// <returns>An <see cref="IQuantity"/> object.</returns>
+        /// <exception cref="System.NotSupportedException">The conversion cannot be performed.</exception>
+        /// <exception cref="ArgumentException">Unit value is not a know unit enum type.</exception>
+        public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo culture, object value)
+        {
+            string stringValue = value as string;
+            object result = null;
+
+            if (!string.IsNullOrEmpty(stringValue))
+            {
+                if (double.TryParse(stringValue, NumberStyles.Any, culture, out double dvalue))
+                {
+                    DefaultUnitAttribute defaultUnit = GetAttribute<DefaultUnitAttribute>(context) ?? new DefaultUnitAttribute(default(TQuantity).Unit);
+
+                    result = Quantity.From(dvalue, defaultUnit.UnitType);
+                }
+                else
+                {
+                    // this should not be part of QuantityTypeConverter. it should rather be part of the parse function
+                    stringValue = stringValue.Replace("^-9", "⁻⁹"); 
```

Comment:
I would be happy to see a pull request on bringing this into the parsing code as you comment here!